gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -191,30 +191,44 @@ func (vm *VMKeeper) AddPackage(ctx sdk.Context, msg MsgAddPackage) error {
 	// Parse and run the files, construct *PV.
 	m2 := gno.NewMachineWithOptions(
 		gno.MachineOptions{
-			PkgPath:   "",
-			Output:    os.Stdout, // XXX
-			Store:     store,
-			Alloc:     store.GetAllocator(),
-			Context:   msgCtx,
-			MaxCycles: vm.maxCycles,
+			PkgPath:    "",
+			Output:     os.Stdout, // XXX
+			Store:      gnostore,
+			Alloc:      gnostore.GetAllocator(),
+			Context:    msgCtx,
+			MaxCycles:  vm.maxCycles,
+			GasMeter: ctx.GasMeter(),
 		})
-	defer m2.Release()
```

Comment:
I don't see why this should be removed. 
Shouldn't this release be deferred first before any panic recover defer below? 
The new code seems more brittle (somebody might forget to release).

Reply:
```go
package main

func main() {

	defer func() {
		println("defer 1")
	}()

	defer func() {
		if r := recover(); r != nil {
			println("defer 2")
			panic("panic defer")
		}
	}()

	panic("panic")

}

// Output:
// defer 2
// defer 1

```

Reply:
It's essentially the same logic since m2 is released in L215. Here's why I made this change: There are two adjacent defer functions with complicated recovery and switching logic in the second defer function. I merged the two defers into one. This way, when people review the code, they can examine a single defer function from the top down without having to remember that the first defer was pushed onto the stack first and executed after the second defer when the first is popped last from the stack upon the function's return.

Reply:
I agree that calling m.Release() right after creating a new machine instance is good practice is valid. The real tricky part, and the place that's easy to forget, lies in error and panic handling within the second defer(). Maybe it's just me, but I feel that managing panic and error handling with two defer statements requires more mental effort than using just one.

Reply:
I misunderstood the suggestion and will update it.